diff --git modules.list modules.list
index 7896479c981..c1a5e664b86 100644
--- modules.list
+++ modules.list
@@ -50,4 +50,7 @@ jdk.xml.dom,
 jdk.zipfs,
 jdk.hotspot.agent,
 jetbrains.api,
-jetbrains.api.impl
+jetbrains.api.impl,
+jcef,
+gluegen.rt,
+jogl.all
diff --git src/java.desktop/share/classes/module-info.java src/java.desktop/share/classes/module-info.java
index 897647ee368..781d1809493 100644
--- src/java.desktop/share/classes/module-info.java
+++ src/java.desktop/share/classes/module-info.java
@@ -116,7 +116,11 @@ module java.desktop {
     // see make/GensrcModuleInfo.gmk
     exports sun.awt to
         jdk.accessibility,
-        jdk.unsupported.desktop;
+        jdk.unsupported.desktop,
+        jcef,
+        jogl.all;
+
+    exports java.awt.peer to jcef;
 
     exports java.awt.dnd.peer to jdk.unsupported.desktop;
     exports sun.awt.dnd to jdk.unsupported.desktop;
