Index: modules.list
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- a/modules.list	(revision e7609c9c21780fef08dbaddd86980806e2463cd9)
+++ b/modules.list	(date 1601891221187)
@@ -61,7 +61,4 @@
 jdk.unsupported,
 jdk.xml.dom,
 jdk.zipfs,
-jdk.hotspot.agent,
-jcef,
-gluegen.rt,
-jogl.all
+jdk.hotspot.agent
Index: src/java.desktop/share/classes/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- a/src/java.desktop/share/classes/module-info.java	(revision e7609c9c21780fef08dbaddd86980806e2463cd9)
+++ b/src/java.desktop/share/classes/module-info.java	(date 1601568912517)
@@ -109,9 +109,7 @@
     // see make/GensrcModuleInfo.gmk
     exports sun.awt to
         jdk.accessibility,
-        jdk.unsupported.desktop,
-        jcef,
-        jogl.all;
+        jdk.unsupported.desktop;

     exports java.awt.peer to jcef;

