Index: modules.list
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- modules.list	(revision 834c441b0f9d1a705a77f67768fa5e0411067b12)
+++ modules.list	(date 1596011136827)
@@ -53,5 +53,4 @@
 jdk.unsupported,
 jdk.xml.dom,
 jdk.zipfs,
-jdk.hotspot.agent,
-jcef
+jdk.hotspot.agent
Index: src/java.desktop/share/classes/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/java.desktop/share/classes/module-info.java	(revision 834c441b0f9d1a705a77f67768fa5e0411067b12)
+++ src/java.desktop/share/classes/module-info.java	(date 1596011136835)
@@ -109,10 +109,7 @@
     // see make/GensrcModuleInfo.gmk
     exports sun.awt to
         jdk.accessibility,
-        jdk.unsupported.desktop,
-        jcef;
-
-    exports java.awt.peer to jcef;
+        jdk.unsupported.desktop;
 
     exports java.awt.dnd.peer to jdk.unsupported.desktop;
     exports sun.awt.dnd to jdk.unsupported.desktop;
