Index: modules.list
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- modules.list	(revision ab073976c4f85f154ff1421b7ac888eee86b6595)
+++ modules.list	(date 1588171510146)
@@ -20,13 +20,6 @@
 java.transaction.xa,
 java.xml,
 java.xml.crypto,
-javafx.base,
-javafx.controls,
-javafx.fxml,
-javafx.graphics,
-javafx.media,
-javafx.swing,
-javafx.web,
 jdk.accessibility,
 jdk.aot,
 jdk.attach,
