Index: modules.list
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- modules.list	(revision 8ff658704ad572469e293cf84de8295a01340673)
+++ modules.list	(date 1592804013230)
@@ -20,13 +20,6 @@
 java.transaction.xa,
 java.xml,
 java.xml.crypto,
-javafx.base,
-javafx.controls,
-javafx.fxml,
-javafx.graphics,
-javafx.media,
-javafx.swing,
-javafx.web,
 jdk.accessibility,
 jdk.aot,
 jdk.attach,
