Index: modules.list
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- modules.list	(revision 57ff1dba25206f9f078b48995aa8744e51b1dd2a)
+++ modules.list	(date 1584500341724)
@@ -20,13 +20,6 @@
 java.transaction.xa,
 java.xml,
 java.xml.crypto,
-javafx.base,
-javafx.controls,
-javafx.fxml,
-javafx.graphics,
-javafx.media,
-javafx.swing,
-javafx.web,
 jdk.accessibility,
 jdk.aot,
 jdk.charsets,
