From 86c27155386c1c40642c99c63a242d1f5d8601a5 Mon Sep 17 00:00:00 2001
From: Vladimir Dvorak <vladimir.dvorak@jetbrains.com>
Date: Sat, 28 Nov 2020 19:31:08 +0100
Subject: [PATCH 28/34] Remove unused fieldType

---
 src/hotspot/share/classfile/vmSymbols.hpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/hotspot/share/classfile/vmSymbols.hpp b/src/hotspot/share/classfile/vmSymbols.hpp
index 6a3b234b222..eb06684a288 100644
--- a/src/hotspot/share/classfile/vmSymbols.hpp
+++ b/src/hotspot/share/classfile/vmSymbols.hpp
@@ -465,7 +465,6 @@
   template(static_offset_name,                        "staticOffset")                             \
   template(static_base_name,                          "staticBase")                               \
   template(field_offset_name,                         "fieldOffset")                              \
-  template(field_type_name,                           "fieldType")                                \
                                                                                                   \
    /* name symbols needed by intrinsics */                                                         \
                                                                                                   \
-- 
2.23.0

